### PR TITLE
(hopefully) fix death position in multiplayer

### PR DESCRIPTION
**Pull request by ekcire06**

### CODE REVIEW

**Comment by ekcire06** [comment]
:|

**Comment by C10udburst** [comment]
I tested it on test.2b2t.org and it doesn't work there

**Comment by ekcire06** [comment]
yeah sorry about that... it should work now.

**Comment by ekcire06** [comment]
the problem was that when you die in a multiplayer server your cords get set to null, and deathcords checks your cords after you die. what I changed was now it records your coordinates whenever you get hit, and when you die it takes the cords of (either the last nonfatal hit you got or the hit that killed you, honestly im not sure) and uses those cords

File: src/main/java/minegame159/meteorclient/modules/player/DeathPosition.java [hunk 1]
```diff
@@ -55,27 +61,32 @@ public DeathPosition() {
...
 
             String time = dateFormat.format(new Date());
-            //ChatUtils.moduleInfo(this, "Died at (highlight)%.0f(default), (highlight)%.0f(default), (highlight)%.0f (default)on (highlight)%s(default).", mc.player.getX(), mc.player.getY(), mc.player.getZ(), time);
-            BaseText msg = new LiteralText("Died at ");
-            msg.append(ChatUtils.formatCoords(mc.player.getPos()));
-            msg.append(".");
-            ChatUtils.moduleInfo(this,msg);
+            ChatUtils.moduleInfo(this, "Died at (highlight)%.0f(default), (highlight)%.0f(default), (highlight)%.0f (default)on (highlight)%s(default).", damagedplayerX, damagedplayerY, damagedplayerZ, time);
+            //BaseText msg = new LiteralText("Died at ");
+            //msg.append(ChatUtils.formatCoords(mc.player.getPos()));
```
Comment (C10udburst) [inline src/main/java/minegame159/meteorclient/modules/player/DeathPosition.java:78]: I don't know what the developers think, but i'd use `msg.append(ChatUtils.formatCoords(new Vec3d(damagedplayerX, damagedplayerY,damagedplayerZ ))))`. The `formatCoords` function allows coords to be clickable in chat (makes baritone path to those coords).

Comment (ekcire06) [reply src/main/java/minegame159/meteorclient/modules/player/DeathPosition.java]: yeah thats a good idea. sorry for being so bad at this.

**Comment by ekcire06** [comment]
im an idiot. wasted 3 hours, and the problem was _I forgot to import vec3d_. im gonna close this pull request and make another one with 1 commit, not 20

**Pull request closed without merging**